stl/inc/vector

```diff
@@ -1761,15 +1634,94 @@ private:
+            _Tidy_guard<vector> _Guard{this};
+            if constexpr (sizeof...(_Val) == 0) {
+                _My_data._Mylast = _Uninitialized_value_construct_n(_My_data._Myfirst, _Count, _Al);
+            } else if constexpr (sizeof...(_Val) == 1) {
+                static_assert(is_same_v<_Valty..., const _Ty&>, "Wrong type passed to _Construct_n");
+                _My_data._Mylast = _Uninitialized_fill_n(_My_data._Myfirst, _Count, _Val..., _Al);
+            } else if constexpr (sizeof...(_Val) == 2) {
+                _My_data._Mylast = _Uninitialized_copy(_STD forward<_Valty>(_Val)..., _My_data._Myfirst, _Al);
+            } else {
+                static_assert(_Always_false<_Ty>, "Should be unreachable");
+            }
+            _Guard._Target = nullptr;
+        }
+
+        _Proxy._Release();
+    }
+
+    _CONSTEXPR20_CONTAINER void _Move_assign_unequal_alloc(vector& _Right) noexcept(
+        !is_same_v<_Choose_pocma<_Alty>, _No_propagate_allocators>) {
+        auto& _Al = _Getal();
+        if constexpr (is_same_v<_Choose_pocma<_Alty>, _Propagate_allocators>) {
+            _Tidy();
+#if _ITERATOR_DEBUG_LEVEL != 0
+            // intentionally slams into noexcept on OOM, TRANSITION, VSO-466800
+            _Mypair._Myval2._Reload_proxy(
+                _GET_PROXY_ALLOCATOR(_Alty, _Al), _GET_PROXY_ALLOCATOR(_Alty, _Right._Getal()));
+#endif
+            _Pocma(_Al, _Right._Getal());
+            _Mypair._Myval2._Take_contents(_Right._Mypair._Myval2);
+        } else {
+            auto& _Right_data    = _Right._Mypair._Myval2;
+            const pointer _First = _Right_data._Myfirst;
+            const pointer _Last  = _Right_data._Mylast;
+            const auto _Newsize  = static_cast<size_type>(_Last - _First);
+
+            auto& _My_data    = _Mypair._Myval2;
+            pointer& _Myfirst = _My_data._Myfirst;
+            pointer& _Mylast  = _My_data._Mylast;
+
+            _My_data._Orphan_all();
+
+            const auto _Oldcapacity = static_cast<size_type>(_My_data._Myend - _Myfirst);
+            if constexpr (conjunction_v<bool_constant<_Ptr_copy_cat<_Ty*, _Ty*>::_Trivially_copyable>,
+                              _Uses_default_construct<_Alty, _Ty*, _Ty>, _Uses_default_destroy<_Alty, _Ty*>>) {
+                if (_Newsize > _Oldcapacity) {
+                    _Clear_and_reserve_geometric(_Newsize);
+                }
+
+                _Mylast = _Refancy<pointer>(_Copy_memmove(_Unfancy(_First), _Unfancy(_Last), _Unfancy(_Myfirst)));
+            } else {
+                auto _Oldsize = static_cast<size_type>(_Mylast - _Myfirst);
+
+                if (_Newsize > _Oldsize) {
+                    if (_Newsize > _Oldcapacity) { // reallocate
+                        _Clear_and_reserve_geometric(_Newsize);
+                        _Oldsize = 0;
+                    }
+
+                    const pointer _Mid = _First + _Oldsize;
+                    _Move_unchecked(_First, _Mid, _Myfirst);

```

Comment:
This is causing a test failure in the "minimal container requirements" test (it appears that this line is attempting to perform move assignment, from a member function which doesn't require move assignable elements).
```
D:\build\out\inc\xutility(4287): error C2248: 'erasable::operator =': cannot access private member declared in class 'erasable'
C:\a\1\s\tests\std\tests\Dev11_0437519_container_requirements\test.compile.pass.cpp(351): note: see declaration of 'erasable::operator ='
C:\a\1\s\tests\std\tests\Dev11_0437519_container_requirements\test.compile.pass.cpp(351): note: see declaration of 'erasable'
D:\build\out\inc\vector(1713): note: see reference to function template instantiation '_OutIt *std::_Move_unchecked<erasable*,erasable*>(_InIt,_InIt,_OutIt)' being compiled
        with
        [
            _OutIt=erasable *,
            _InIt=erasable *
        ]
D:\build\out\inc\vector(1672): note: while compiling class template member function 'void std::vector<V,A>::_Move_assign_unequal_alloc(std::vector<V,A> &) noexcept'
        with
        [
            V=erasable,
            A=pocma_allocator<erasable>
        ]
D:\build\out\inc\vector(583): note: see reference to function template instantiation 'void std::vector<V,A>::_Move_assign_unequal_alloc(std::vector<V,A> &) noexcept' being compiled
        with
        [
            V=erasable,
            A=pocma_allocator<erasable>
        ]
C:\a\1\s\tests\std\tests\Dev11_0437519_container_requirements\test.compile.pass.cpp(1042): note: see reference to class template instantiation 'std::vector<V,A>' being compiled
        with
        [
            V=erasable,
            A=pocma_allocator<erasable>
        ]
C:\a\1\s\tests\std\tests\Dev11_0437519_container_requirements\test.compile.pass.cpp(1042): note: while compiling class template member function 'test_allocator_aware_assignment_from_rvalue_pocma<tag_vector>::test_allocator_aware_assignment_from_rvalue_pocma(void)'
C:\a\1\s\tests\std\tests\Dev11_0437519_container_requirements\test.compile.pass.cpp(1067): note: see reference to function template instantiation 'test_allocator_aware_assignment_from_rvalue_pocma<tag_vector>::test_allocator_aware_assignment_from_rvalue_pocma(void)' being compiled
C:\a\1\s\tests\std\tests\Dev11_0437519_container_requirements\test.compile.pass.cpp(1067): note: see reference to class template instantiation 'test_allocator_aware_assignment_from_rvalue_pocma<tag_vector>' being compiled
C:\a\1\s\tests\std\tests\Dev11_0437519_container_requirements\test.compile.pass.cpp(3017): note: see reference to function template instantiation 'void check_all_allocator_aware_requirements<tag_vector>(void)' being compiled
C:\a\1\s\tests\std\tests\Dev11_0437519_container_requirements\test.compile.pass.cpp(3033): note: see reference to function template instantiation 'void assert_container<tag_vector>(void)' being compiled
```
There are other failing tests.

Reply:
Found the bugs